### DISCUSSION TITLE
S2BufferOperation calculation results are not as expected

### DISCUSSION THREAD

**Discussion by LemonLiTree** [google/s2geometry#319] [Q&A]
I want to get a buffer with a specified point and a specified radius，My code is as follows

```c++
    double lat = 100;
    double lng = 2;
    double buffer_radius = 50;
    S2BufferOperation::Options options;
    options.set_buffer_radius(S1Angle::Radians(buffer_radius));//specified buffer_radius
    auto output = std::make_unique<S2Polygon>();
    S2BufferOperation op(std::make_unique<s2builderutil::S2PolygonLayer>(output.get()), options);
    op.AddPoint(S2LatLng::FromDegrees(lat, lng).ToPoint());
    S2Error error;
    if(!op.Build(&error)){
            return nullptr;
      }
```
There is a loop in the output output, this loop num_vertices = 1; 
Is there something wrong with my code?

**Comment by smcallis** [comment]
You're buffering with 50 radians so it's probably generating a full polygon
for you.   Maybe try S1Angle::Degrees(buffer_radius)?  Also is there a
reason S2Cap won't work?

**Comment by LemonLiTree** [comment]
I try to S2Earth::MetersToAngle(buffer_radius) 
and add 
```c++
options.set_circle_segments(8);//
```
the result is correct.
